test/lint/streams.rb

```diff
@@ -346,6 +346,20 @@ def test_xread_with_block_option
       assert_equal({}, actual)
     end
 
+    def test_xread_does_not_raise_timeout_error_when_the_block_option_is_zero_msec
+      prepared = false
+      actual = nil
+      wire = Wire.new do
+        prepared = true
+        actual = redis.xread('s1', 0, block: 0)
+      end
+      Wire.pass until prepared
+      redis.dup.xadd('s1', { f: 'v1' }, id: '0-1')
```

Comment:
Not sure if `.dup` is a good idea. It's only a shallow clone, so you might be sharing the mutex etc.

Reply:
Hello,

I appreciate for your review. 

It seems that the [`.dup`](https://github.com/redis/redis-rb/blob/v4.1.0/lib/redis.rb#L3303-L3305) is overridden. It looks like that the [`.dup`](https://github.com/redis/redis-rb/blob/v4.1.0/lib/redis.rb#L3303-L3305) make another instance.

Reply:
TIL.

LGTM then.